pkg/cloud/services/networking/port.go

```diff
@@ -166,6 +167,14 @@ func (s *Service) GetOrCreatePort(eventObject runtime.Object, clusterName string
 		return nil, err
 	}
 
+	defer func() {
+		if err != nil {
+			if err := s.DeletePort(eventObject, port.ID); err != nil {
+				record.Warnf(eventObject, "FailedDeletePort", "Failed to delete port %s: %v", portName, err)
+			}
+		}
+	}()
+
```

Comment:
This is an unfortunate wart, because deleting un-tracked ports in a defer block is what we're trying to avoid here.

I think you can do away with the need for it in this function if on line 70 above we just skip port creation if the port already exists, and then do everything below. I think everything below is idempotent, so it should be ok if we do it more than once.